File: capco-noc-2020-fe/src/app/_models/transaction.ts
import {User} from "./user";

export class Transaction {

  id: number;
  receiver: User;
  payer: User;
  amount: number;
  date: Date;

  // Get new transaction object from JSON values
  public static parseFromJson(id, receiver, payer, date, amount): Transaction {
    let newTransaction = new Transaction();
    newTransaction.id = id;
    newTransaction.receiver = receiver;
    newTransaction.payer = payer;
    newTransaction.amount = amount;
    newTransaction.date = new Date(date);
    return newTransaction;
  }
}

File: capco-noc-2020-be/src/main/java/com/capco/noc2020/entity/Transaction.java
package com.capco.noc2020.entity;

import lombok.Data;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@ToString
@Entity(name = "transaction")
public class Transaction {
    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime date;
    @ManyToOne
    private User payer;
    @ManyToOne
    private User receiver;
    private BigDecimal amount;
}
